src/morgenstern.jl

```diff
@@ -1,108 +1,311 @@
 """
-Randomly sample for ε∈𝔽q until you find one such that x^2+x+ε∈𝔽q[x] is irreducible.
+Finds an irreducible polynomial of the form ``f(x) = x^2 + x + \\varepsilon`` over
+the finite field ``\\mathbb{F}_q``.
 
-Takes as input 𝔽q[x], the polynomial ring we want to work with.
+For quadratic polynomials, irreducibility is equivalent to having no roots in the base field:
 
-Returns x^2+x+ε.
+```jldoctest
+julia> using Oscar
 
-See [morgenstern1994existence](@cite).
+julia> 𝔽₂, _ = finite_field(2, 1, "a");
+
+julia> R, x = polynomial_ring(𝔽₂, "x");
+
+julia> f = x^2 + x + 𝔽₂(1);
+
+julia> is_irreducible(f) && isempty(roots(f))
+true
+
+julia> g = x^2 + x + 𝔽₂(0);
+
+julia> is_irreducible(g) || !isempty(roots(g))
+true
+```
+
+In [characteristic](https://en.wikipedia.org/wiki/Characteristic_(algebra)) 2, the standard quadratic form
+``x^2 + \\varepsilon`` is insufficient because its derivative is zero, making it inseparable if it has a root.
+The form ``x^2 + x + \\varepsilon`` is used instead, as its derivative is 1, ensuring separability. Its irreducibility
+over ``\\mathbb{F}_q`` is equivalent to it having no roots in ``\\mathbb{F}_q``.
```

Comment:
This paragraph might be true, but neither one of us understands this paragraph in detail, so let's not risk including documentation that is our own rephrasing of properties we do not fully understand.